sanic/exceptions.py

```diff
@@ -160,7 +163,20 @@ def _render_traceback_html(self, exception, request):
             uri=request.url)
 
     def add(self, exception, handler):
-        self.handlers[exception] = handler
+        self.handlers.append((exception, handler))
+
+    def lookup(self, exception):
+        handler = self.cached_handlers.get(exception, None)
+        if handler is None:
+            for exception_class, handler in self.handlers:
+                if isinstance(exception, exception_class):
+                    self.cached_handlers[type(exception)] = handler
+                    return handler
+            handler = self._missing
```

Comment:
The handler can be `self.default` there, but `_missing` is not replacable by `object()`. In that case, `object()` will be created everytime whenever caching miss.